src/app/ReadClient.h

```diff
@@ -514,6 +518,7 @@ class ReadClient : public Messaging::ExchangeDelegate
         AwaitingSubscribeResponse, ///< The client is waiting for subscribe response
         SubscriptionActive,        ///< The client is maintaining subscription
         InactiveICDSubscription,   ///< The client is waiting to resubscribe for LIT device
+        FailedICDSubscription,     ///< The client is failing during case establishment or subscription priming stage for LIT device
```

Comment:
Why is this part ICD-specific?  Shouldn't we just move into a "failed" state if we failed?  If we then get a check-in while in failed state, we immediately retry....

But also, how is this state different from just having `mIsResubscriptionScheduled` be true?  As in, in this new setup shouldn't receipt of a check-in message when we are not in InactiveICDSubscription just call `TriggerResubscribeIfScheduled`, possibly after resetting backoff bits?

It worries me a lot that we have a bunch of ICD special-casing and complexity here, when we have existing codepaths to do _exactly_ that thing for non-ICDs.  Why does that path not work for ICDs?

Reply:
Yes, I agree, I have removed this state, and it is not necessary..